packages/app/src/systems/Asset/components/AssetSelect/AssetSelect.tsx

```diff
@@ -0,0 +1,243 @@
+import { cssObj } from '@fuel-ui/css';
+import type { DropdownProps } from '@fuel-ui/react';
+import {
+  Avatar,
+  Box,
+  Button,
+  Dropdown,
+  Flex,
+  Icon,
+  IconButton,
+  Stack,
+  Text,
+  Tooltip,
+} from '@fuel-ui/react';
+import type { AssetAmount, Coin } from '@fuel-wallet/types';
+import { bn } from 'fuels';
+import React, { useState } from 'react';
+
+import { getAssetInfoById } from '../../utils';
+
+import { MAX_FRACTION_DIGITS } from '~/config';
+import type { Maybe } from '~/systems/Core';
+import type { TxInputCoin, TxOutputCoin } from '~/systems/Transaction';
+
+export type AssetSelectInput = AssetAmount | Coin | TxOutputCoin | TxInputCoin;
+
+export type AssetSelectProps = DropdownProps & {
+  items?: Maybe<AssetSelectInput[]>;
+  onSelect: (asset?: AssetSelectInput | null) => void;
+  selected?: Maybe<AssetSelectInput>;
+};
+
+export function AssetSelect({
+  items,
+  selected,
+  onSelect,
+  ...props
+}: AssetSelectProps) {
+  /** i'm using a ref here instead of a state in order to don't trigger a new re-render */
+  const [width, setWidth] = useState<Maybe<number>>(null);
```

Comment:
I think we should keep the menu with a hard width, reducing the new for dynamic control here. Any issues on the component that requires us to use dynamic width?